core/org.eclipse.cdt.core/parser/org/eclipse/cdt/internal/core/dom/parser/ASTAmbiguousBinaryVsCastExpression.java

```diff
@@ -224,19 +224,21 @@ private int getPrecendence(IASTBinaryExpression r) {
 		case IASTBinaryExpression.op_max:
 		case IASTBinaryExpression.op_min:
 			return 7;
+		case IASTBinaryExpression.op_threewaycomparison:
+			return 8;
 		case IASTBinaryExpression.op_shiftLeft:
 		case IASTBinaryExpression.op_shiftRight:
-			return 8;
```

Comment:
I struggled a little bit to understand the use case of the binary vs cast ambiguation, as I couldn't come up with any examples to test this. However your changes look reasonable, you just added <=> and gave it a higher precedence.